drivers/espi/espi_it8xxx2.c

```diff
@@ -153,6 +155,97 @@ static const struct ec2i_t pmc1_settings[] = {
+	{HOST_INDEX_LDN, LDN_PMC2},
+	/* I/O Port Base Address (data/command ports) */
+	{HOST_INDEX_IOBAD0_MSB, IT8XXX2_ESPI_HC_DATA_PORT_MSB},
+	{HOST_INDEX_IOBAD0_LSB, IT8XXX2_ESPI_HC_DATA_PORT_LSB},
+	{HOST_INDEX_IOBAD1_MSB, IT8XXX2_ESPI_HC_CMD_PORT_MSB},
+	{HOST_INDEX_IOBAD1_LSB, IT8XXX2_ESPI_HC_CMD_PORT_LSB},
+	/* Set IRQ=00h for logical device */
+	{HOST_INDEX_IRQNUMX, 0x00},
+	/* Enable logical device */
+	{HOST_INDEX_LDA, 0x01},
+};
+#endif
+
+#if defined(CONFIG_ESPI_PERIPHERAL_EC_HOST_CMD) || \
+	defined(CONFIG_ESPI_PERIPHERAL_ACPI_SHM_REGION)
+/* Host to RAM (H2RAM) memory mapping, 4K aligned */
+#define IT8XXX2_ESPI_H2RAM_POOL_SIZE 0x1000
+static uint8_t h2ram_pool[IT8XXX2_ESPI_H2RAM_POOL_SIZE] __aligned(0x1000);
+
+#if defined(CONFIG_ESPI_PERIPHERAL_ACPI_SHM_REGION) && \
+	((CONFIG_ESPI_PERIPHERAL_ACPI_SHM_REGION_PORT_NUM + \
+	CONFIG_ESPI_IT8XXX2_ACPI_SHM_H2RAM_SIZE) > IT8XXX2_ESPI_H2RAM_POOL_SIZE)
+#error "ACPI shared memory region out of h2ram"
+#endif
+
+#if defined(CONFIG_ESPI_PERIPHERAL_EC_HOST_CMD) && \
+	((CONFIG_ESPI_PERIPHERAL_HOST_CMD_PARAM_PORT_NUM + \
+	CONFIG_ESPI_IT8XXX2_HC_H2RAM_SIZE) > IT8XXX2_ESPI_H2RAM_POOL_SIZE)
+#error "EC host command parameters out of h2ram"
+#endif
+
+static const struct ec2i_t smfi_settings[] = {
+	/* Select logical device 0Fh(SMFI) */
+	{HOST_INDEX_LDN, LDN_SMFI},
+	/* Internal RAM base address on eSPI I/O space */
+	{HOST_INDEX_DSLDC6, 0x00},
+	/* Enable H2RAM eSPI I/O cycle */
+	{HOST_INDEX_DSLDC7, 0x01},
+	/* Enable logical device */
+	{HOST_INDEX_LDA, 0x01},
+};
+
+static void smfi_it8xxx2_init(const struct device *dev)
+{
+	const struct espi_it8xxx2_config *const config = dev->config;
+	struct flash_it8xxx2_regs *const smfi_reg =
+		(struct flash_it8xxx2_regs *)config->base_smfi;
+	struct gctrl_it8xxx2_regs *const gctrl = ESPI_IT8XXX2_GET_GCTRL_BASE;
+	uint8_t h2ram_offset;
+
+	/* Set the host to RAM cycle address offset */
+	h2ram_offset = ((uint32_t)h2ram_pool & 0xffff) /
+				IT8XXX2_ESPI_H2RAM_POOL_SIZE;
+	gctrl->GCTRL_H2ROFSR |= h2ram_offset;
+
+#ifdef CONFIG_ESPI_PERIPHERAL_EC_HOST_CMD
+	/* Set host RAM window 0 base address */
+	smfi_reg->SMFI_HRAMW0BA =
+		(CONFIG_ESPI_PERIPHERAL_HOST_CMD_PARAM_PORT_NUM >> 4) & 0xff;
+	/* Set host RAM window 0 size. (allow R/W) */
+	smfi_reg->SMFI_HRAMW0AAS =
+		find_msb_set(CONFIG_ESPI_IT8XXX2_HC_H2RAM_SIZE / 16) - 1;
+	/* Enable window 0, H2RAM through IO cycle */
+	smfi_reg->SMFI_HRAMWC |= (SMFI_H2RAMPS | SMFI_H2RAMW0E);
+#endif
+
+#ifdef CONFIG_ESPI_PERIPHERAL_ACPI_SHM_REGION
+	/* Set host RAM window 1 base address */
+	smfi_reg->SMFI_HRAMW1BA =
+		(CONFIG_ESPI_PERIPHERAL_ACPI_SHM_REGION_PORT_NUM >> 4) & 0xff;
+	/* Set host RAM window 1 size. (read-only) */
+	smfi_reg->SMFI_HRAMW1AAS =
+		find_msb_set(CONFIG_ESPI_IT8XXX2_ACPI_SHM_H2RAM_SIZE / 16) - 1;
+	smfi_reg->SMFI_HRAMW1AAS |= SMFI_HRAMWXWPE_ALL;
+	/* Enable window 1, H2RAM through IO cycle */
+	smfi_reg->SMFI_HRAMWC |= (SMFI_H2RAMPS | SMFI_H2RAMW1E);
+#endif
+}
+#endif
```

Comment:
Document which #if is closed here.  I think `defined(CONFIG_ESPI_PERIPHERAL_EC_HOST_CMD) || defined(CONFIG_ESPI_PERIPHERAL_ACPI_SHM_REGION)`

Reply:
Done.